pkg/apis/pipeline/v1beta1/task_validation.go

```diff
@@ -351,9 +352,10 @@ func ValidateParameterVariables(ctx context.Context, steps []Step, params []Para
 			stringParameterNames.Insert(p.Name)
 		}
 	}
-
 	errs = errs.Also(validateNameFormat(stringParameterNames.Insert(arrayParameterNames.List()...), objectParamSpecs))
-	errs = errs.Also(validateVariables(ctx, steps, "params", allParameterNames))
+	if !skipVariableValidation {
+		errs = errs.Also(validateVariables(ctx, steps, "params", allParameterNames))
+	}
```

Comment:
just curious if we do so, are we going to skip the following  validateArrayUsage, validateObjectDefault as well?

Reply:
I don't think so. Those checks are done separately I think.